File: neoforge/src/main/java/dev/wren/truss/platform/NeoForgePlatformHelper.java
package dev.wren.truss.platform;

import dev.wren.truss.platform.services.PlatformHelper;
import net.neoforged.fml.loading.FMLLoader;

public class NeoForgePlatformHelper implements PlatformHelper {
    @Override
    public boolean isClient() {
        return FMLLoader.getDist().isClient();
    }

    @Override
    public boolean isDedicatedServer() {
        return FMLLoader.getDist().isClient();
    }

    @Override
    public String getDist() {
        return FMLLoader.getDist().name();
    }

    @Override
    public Platform getPlatform() {
        return Platform.NEOFORGE;
    }
}

File: neoforge/src/main/kotlin/dev/wren/truss/platform/NeoForgePlatformHelper.kt
package dev.wren.truss.platform

import dev.wren.truss.platform.services.PlatformHelper
import net.neoforged.fml.loading.FMLLoader

class NeoForgePlatformHelper : PlatformHelper {
    override val isClient: Boolean = FMLLoader.getDist().isClient
    override val isDedicatedServer: Boolean = FMLLoader.getDist().isDedicatedServer
    override val dist: String = FMLLoader.getDist().name
    override val platform: PlatformHelper.Platform = PlatformHelper.Platform.FORGE
}